### PR TITLE
Shadowling Glare nerf

### PR DESCRIPTION
**Pull request by ghost**
Glare and Lesser Glare have no effect on targets that are blind. (It still goes on cooldown as you attempt to blind them.)
Lesser Glare is now countered by anything that gives eye protection (Welding goggles/masks, sunglasses and sechuds.)

### CODE REVIEW

**Comment by TheOneAndOnlyCreeperJoe** [comment]
Shadowlings are much deserving of a nerf.
Finally someone stepped up and did it.

**Comment by ghost** [comment]
I hardly call the blind being immune a nerf but yeah. I've been cucked too many times from a shitty thrall glaring me as I stun him. If you want to take on the big boys, you better come more prepared and without the "I can stun you while I'm stunned" power.

**Comment by Spacedong** [comment]
neat

**Comment by ghost** [comment]
Also should mention that the eye implants also give Lesser Glare protection. I didn't really want to snowflake code against it but it kinda makes sense??

**Comment by Crystalwarrior** [comment]
This honestly makes sense, from both consistency and gameplay standpoints.
:+1:

**Comment by ghost** [comment]
I'm not going to +1 my own PR.
GMs, get your asses in here.

**Comment by calibraptor** [comment]
+1

Take shadowlings down a peg

**Comment by Lawligagger** [comment]
Seems good to me hombre

**Comment by dreamcrab** [comment]
yeah, i like the lesser glare nerf. +1

**Comment by ghost** [comment]
2 headmin approval, IDK if crystal giving that thumbs up was a code check or a "this is a good idea" thumbs up though.

**Comment by JohnGinnane** [comment]
Code checking it now

File: code/game/gamemodes/shadowling/shadowling_abilities.dm [hunk 1]
```diff
@@ -38,6 +38,8 @@
 			return
 		var/mob/living/carbon/human/M = target
 		user.visible_message("<span class='warning'><b>[user]'s eyes flash a blinding red!</b></span>")
+		if((target.disabilities & BLIND) | target.eye_blind)
```
Comment (JohnGinnane) [inline code/game/gamemodes/shadowling/shadowling_abilities.dm]: Should this not be ||?

File: code/game/gamemodes/shadowling/shadowling_abilities.dm [hunk 2]
```diff
@@ -681,8 +683,14 @@ datum/reagent/shadowling_blindness_smoke //Reagent used for above spell
 			user << "<span class='warning'>You cannot glare at allies!</span>"
 			revert_cast()
 			return
+		if(target.check_eye_prot() > 0)
+			user << "<span class='warning'>You can't get a direct gaze on [target]'s eyes!</span>"
+			revert_cast()
+			return
 		var/mob/living/carbon/human/M = target
 		user.visible_message("<span class='warning'><b>[user]'s eyes flash a bright red!</b></span>")
+		if((target.disabilities & BLIND) | target.eye_blind)
```
Comment (JohnGinnane) [inline code/game/gamemodes/shadowling/shadowling_abilities.dm]: Same as above

**Comment by JohnGinnane** [comment]
Code looks good, but the OR operator when checking for blindness threw me off

**Comment by ghost** [comment]
|| is java, I think byond just uses |

**Comment by Crystalwarrior** [comment]
Byond is || you nerd, check literally everywhere else in code :mememan3:

**Comment by ghost** [comment]
I never fucked it up lads

**Comment by Crystalwarrior** [comment]
Fix it, | is a bitwise operator in BYOND.

**Comment by ghost** [comment]
https://github.com/HippieStationCode/HippieStation13/pull/2261/commits/b14f8d0f40756d83c4e9802bcf76452b567b598d
I NEVER FUCKED IT UP LADS

**Comment by Crystalwarrior** [comment]
laddie I'll have your ass on a silver platter

**Comment by JohnGinnane** [comment]
@Crystalwarrior the condition would work would it not though?

**Comment by ghost** [comment]
god damn it crystal, your eyes should let you know that I have not made an error? look at the code, there is || operators and always have been???

**Comment by nalarmg** [comment]
http://www.byond.com/docs/guide/chap06.html#6.11.3

| is the correct OR operator for bitwise operations. || is just a generic OR operator.

**Comment by ghost** [comment]
I'm pulling his chain Nalar, I fixed it to || in my last commit.

**Comment by Crystalwarrior** [comment]
Okok merging

**Pull request merged**